### DISCUSSION TITLE
No authority - when trying to get API

### DISCUSSION THREAD

**Discussion by ckortenhorst** [hultenvp/solis-sensor#207] [General]
Anyone else experiencing issues creating API key on solis cloud website https://www.soliscloud.com/#/apiManage 

I click on agree but get following 

<img width="428" alt="image" src="https://user-images.githubusercontent.com/5760635/200693136-8e84879b-3404-4bdb-b455-6c8cab52bbe1.png">

<img width="464" alt="image" src="https://user-images.githubusercontent.com/5760635/200693152-d8f59feb-6fb6-44ef-b104-ff0d9ef46f52.png">

**Comment by peledudok** [comment]
As far as i know you should first sent an email to request for api.
I also saw on the app I'm announcement for maintenance 9/11 utc from 2.00 till...

**Reply by ckortenhorst** [reply]
Any idea who we should email?

**Reply by peledudok** [reply]
https://solis-service.solisinverters.com/support/solutions/articles/44002212561-api-access-soliscloud

**Reply by hultenvp** [reply]
@ckortenhorst : let me know if it works. Would like to capture the new/correct procedure in the readme

**Reply by ckortenhorst** [reply]
Seams you have to log ticket just done it now will revert back when I know more.

**Reply by grbasta** [reply]
Good morning after opening a ticket on https://solis-service.solisinverters.com/support/home 3 days ago, this morning by chance without having had a response to the ticket, I was able to activate the API.
![image](https://user-images.githubusercontent.com/46572022/201294473-c236602d-ce18-4ef5-92fd-7ea48138ca25.png)
Until yesterday, the tab circled in red did not appear and this morning I activated and loaded integration on Home Assistant

**Reply by peledudok** [reply]
and did it work? 
because i got it on the same way but the intergration is causing me a headdeck because i can't get it to work

**Reply by grbasta** [reply]
Are you sure you have entered everything correctly?
it worked for me at first sight

**Reply by peledudok** [reply]
<img width="1021" alt="image" src="https://user-images.githubusercontent.com/88967555/201344602-54b6193e-5b41-4024-adcc-67833d5869ac.png">

I fill in the following credentials first popup :
<img width="269" alt="image" src="https://user-images.githubusercontent.com/88967555/201345641-1ad8542a-7847-4448-9739-ebd78d596e58.png">
I'm not sure what to fill in at "the name of the inverter" field I tried Solis (defaultname), my own plant name "simpelzon" , my user name "simpel"

Second popup:
<img width="261" alt="image" src="https://user-images.githubusercontent.com/88967555/201346037-763329be-4519-4834-b30f-621d065cdea7.png">
at "portal username or email adres" i tried my email adres and my username.

after submit i get:
<img width="266" alt="image" src="https://user-images.githubusercontent.com/88967555/201346347-fe11aba1-09db-4b15-a8f5-ece9ae6eacff.png">

and the log said:
2022-11-11 14:04:45.371 DEBUG (MainThread) [custom_components.solis.soliscloud_api] workarounds: {'correct_daily_on_grid_energy_enabled': True}
2022-11-11 14:04:45.553 INFO (MainThread) [custom_components.solis.soliscloud_api] Login successful
2022-11-11 14:04:45.553 DEBUG (MainThread) [custom_components.solis.soliscloud_api] Found inverters: ['1002010226020296']
2022-11-11 14:04:45.553 DEBUG (MainThread) [custom_components.solis.soliscloud_api] Fetching data for serial: 1002010226020296
2022-11-11 14:04:45.647 INFO (MainThread) [custom_components.solis.soliscloud_api] /v1/api/stationDetail responded with error: 1:数据异常 请联系管理员
2022-11-11 14:04:45.647 DEBUG (MainThread) [custom_components.solis.ginlong_base] AttributeError, plantName does not exist
2022-11-11 14:04:45.647 INFO (MainThread) [custom_components.solis.soliscloud_api] Failed to acquire plant name, login failed

I'm so sad for more then 1 week i'm trying without any luck

**Reply by grbasta** [reply]
![image](https://user-images.githubusercontent.com/46572022/201349799-fba37382-64ea-4125-80e0-8a190d421473.png)
On the soliscloud portal are you as Owner and that the registration email to the portal matches?

**Reply by peledudok** [reply]
i see this:
<img width="626" alt="image" src="https://user-images.githubusercontent.com/88967555/201353789-05beaa19-90a9-49d2-aa46-807dc0786e01.png">

it is in dutch but user type : owner
my username is: simpel
and email : b.manglie@upcmail.nl 
as you can see what i fill in is correct :)

I don't see what i am doing wrong.

**Reply by hultenvp** [reply]
What's really odd is that the integration *can* find the inverter ID, so the first API call does work, suggesting your key/secret settings are good.
Then the second call still fails, like we discussed before. For some reason your setting does not allow to retrieve station details.
So part of the API works, the other part does not. The integration will try to log in endlessly and will fail every time.

I honestly have no clue why you get the "Error 1", other than something's wrong with your account. I think you need to log an issue with Solis to get this figured out.
Actually, if I translate the chinese message to dutch it says you should ;-)

<img width="497" alt="image" src="https://user-images.githubusercontent.com/61835400/201360317-d768e74b-a98c-4a3c-98ca-ea4228e57ab5.png">

Good luck, and if you need help do not hesitate to ask.

**Reply by peledudok** [reply]
thx for helping even the issue is not solved. Lets just wait for an answer from Solis support team (ticket open for 5 days now)

**Reply by jmason** [reply]
I have a ticket open for 6 days so far.  I still get the "No Authority" error when I try the webpage.

**Reply by peledudok** [reply]
perhaps there are still issues
